src/IAM/Domain/User/Model/UserRepository.php

```diff
@@ -0,0 +1,12 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Twitter\IAM\Domain\User\Model;
+
+interface UserRepository
+{
+    public function add(User $user): void;
+
+    public function existsByEmail(string $email): bool;
```

Comment:
@adnlv (low) Can be better with `Email` value object.
```suggestion
    public function existsByEmail(Email $email): bool;
```